adapters/33across/33across.go

```diff
@@ -81,63 +104,84 @@ func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.Request
 		Uri:     a.endpoint,
 		Body:    reqJSON,
 		Headers: headers,
-	}, errs
+	}, nil
 }
 
-// Mutate the request to get it ready to send to ttx.
-func preprocess(request *openrtb.BidRequest) error {
-	var imp = &request.Imp[0]
+func makeImps(imp openrtb.Imp) (openrtb.Imp, error) {
+	if imp.Banner == nil && imp.Video == nil {
+		return openrtb.Imp{}, &errortypes.BadInput{
+			Message: fmt.Sprintf("Imp ID %s must have at least one of [Banner, Video] defined", imp.ID),
+		}
+	}
+
 	var bidderExt adapters.ExtImpBidder
 	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-		return &errortypes.BadInput{
+		return openrtb.Imp{}, &errortypes.BadInput{
 			Message: err.Error(),
 		}
 	}
 
 	var ttxExt openrtb_ext.ExtImp33across
 	if err := json.Unmarshal(bidderExt.Bidder, &ttxExt); err != nil {
-		return &errortypes.BadInput{
+		return openrtb.Imp{}, &errortypes.BadInput{
 			Message: err.Error(),
 		}
 	}
 
 	var impExt Ext
 	impExt.Ttx.Prod = ttxExt.ProductId
 
-	// Add zoneid if it's defined
+	impExt.Ttx.Zoneid = ttxExt.SiteId
+
 	if len(ttxExt.ZoneId) > 0 {
 		impExt.Ttx.Zoneid = ttxExt.ZoneId
 	}
 
 	impExtJSON, err := json.Marshal(impExt)
 	if err != nil {
-		return &errortypes.BadInput{
+		return openrtb.Imp{}, &errortypes.BadInput{
 			Message: err.Error(),
 		}
 	}
 
 	imp.Ext = impExtJSON
 
-	if request.Site != nil {
-		siteCopy := *request.Site
-		siteCopy.ID = ttxExt.SiteId
-		request.Site = &siteCopy
-	}
-
 	// Validate Video if it exists
 	if imp.Video != nil {
 		videoCopy, err := validateVideoParams(imp.Video, impExt.Ttx.Prod)
 
 		imp.Video = videoCopy
 
 		if err != nil {
-			return &errortypes.BadInput{
+			return imp, &errortypes.BadInput{
```

Comment:
Should this return statement be returning an empty `imp` object/nil? In case of error, you anyway discard the imp and move on to the next one